tests/test_pagure.py

```diff
@@ -0,0 +1,701 @@
+        self.assertEqual(pagure.origin, 'https://pagure.io/Project-example')
+        self.assertEqual(pagure.tag, 'https://pagure.io/Project-example')
+        self.assertEqual(pagure.api_token, 'aaa')
+
+        # Empty value generates a None API token
+        pagure = Pagure(repository='Project-example', tag='test')
+        self.assertEqual(pagure.repository, 'Project-example')
+        self.assertIsNone(pagure.namespace)
+        self.assertEqual(pagure.origin, 'https://pagure.io/Project-example')
+        self.assertEqual(pagure.tag, 'test')
+        self.assertIsNone(pagure.api_token)
+
+        # Testing initialization when repository is within a namespace
+        pagure = Pagure(namespace='Test-group', repository='Project-example-namespace', api_token='aaa', tag='testing')
+        self.assertEqual(pagure.repository, 'Project-example-namespace')
+        self.assertEqual(pagure.namespace, 'Test-group')
+        self.assertEqual(pagure.origin, 'https://pagure.io/Test-group/Project-example-namespace')
+        self.assertEqual(pagure.tag, 'testing')
+        self.assertEqual(pagure.max_items, MAX_CATEGORY_ITEMS_PER_PAGE)
+        self.assertEqual(pagure.categories, [CATEGORY_ISSUE])
+        self.assertEqual(pagure.api_token, 'aaa')
+        self.assertTrue(pagure.ssl_verify)
+
+    def test_has_resuming(self):
+        """Test if it returns True when has_resuming is called"""
+
+        self.assertEqual(Pagure.has_resuming(), True)
+
+    def test_has_archiving(self):
+        """Test if it returns True when has_archiving is called"""
+
+        self.assertEqual(Pagure.has_archiving(), True)
+
+    @httpretty.activate
+    def test_fetch_issues(self):
+        """Test whether a list of issues is returned"""
+
+        body = read_file('data/pagure/pagure_repo_issue_1')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=body,
+                               status=200,
+                               )
+        pagure = Pagure(repository='Project-example', api_token='aaa')
+        issues = [issues for issues in pagure.fetch(from_date=None, to_date=None)]
+
+        self.assertEqual(len(issues), 1)
+
+        issue = issues[0]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['uuid'], '41071b08dd75f34ca92c6d5ecb844e7a3e5939c6')
+        self.assertEqual(issue['updated_on'], 1583508642.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Project-example')
+        self.assertEqual(len(issue['data']['comments']), 1)
+        self.assertEqual(issue['data']['comments'][0]['user']['name'], 'animeshk08')
+        self.assertEqual(len(issue['data']['comments'][0]['reactions']), 0)
+
+    @httpretty.activate
+    def test_fetch_issues_disabled(self):
+        """Test whether an empty response is returned in case a repository
```

Comment:
Probably this test could be completed to check the text of the warning message (see ref: https://github.com/chaoss/grimoirelab-perceval/blob/28d4b13fdeddaa43c89fd69a530242b1396208d6/tests/test_jenkins.py#L257). 

I would change the description with something like: `Test whether a warning message is logged when the issue tracker is disabled`.